lib/sinatra/base.rb

```diff
@@ -1261,7 +1261,19 @@ def error_block!(key, *block_params)
     end
 
     def dump_errors!(boom)
-      msg = ["#{Time.now.strftime('%Y-%m-%d %H:%M:%S')} - #{boom.class} - #{boom.message}:", *boom.backtrace].join("\n\t")
+      if boom.respond_to?(:detailed_message)
+        msg = boom.detailed_message(highlight: false)
+        if msg =~ /\A(.*?)(?: \(#{ Regexp.quote(boom.class.to_s) }\))?\n/
+          msg = $1
+          additional_msg = $'.lines.map {|s| s.chomp }
```

Comment:
Or this? (The `chomp:` argument should be supported on all Rubies that have `#detailed_message`)

```suggestion
          additional_msg = $'.lines(chomp: true)
```